builtin/providers/azurerm/resource_arm_sql_elasticpool.go

```diff
@@ -0,0 +1,217 @@
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"server_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"edition": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateSqlElasticPoolEdition(),
+			},
+
+			"dtu": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+
+			"db_dtu_min": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+
+			"db_dtu_max": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+
+			"pool_size": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+
+			"creation_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmSqlElasticPoolCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	elasticPoolsClient := client.sqlElasticPoolsClient
+
+	log.Printf("[INFO] preparing arguments for Azure ARM SQL ElasticPool creation.")
+
+	name := d.Get("name").(string)
+	serverName := d.Get("server_name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	tags := d.Get("tags").(map[string]interface{})
+
+	elasticPool := sql.ElasticPool{
+		Name:                  &name,
+		Location:              &location,
+		ElasticPoolProperties: getArmSqlElasticPoolProperties(d),
+		Tags: expandTags(tags),
+	}
+
+	_, err := elasticPoolsClient.CreateOrUpdate(resGroup, serverName, name, elasticPool, make(chan struct{}))
+	if err != nil {
+		return err
+	}
+
+	read, err := elasticPoolsClient.Get(resGroup, serverName, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read SQL ElasticPool %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmSqlElasticPoolRead(d, meta)
+}
+
+func resourceArmSqlElasticPoolRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	elasticPoolsClient := client.sqlElasticPoolsClient
+
+	resGroup, serverName, name, err := parseArmSqlElasticPoolId(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := elasticPoolsClient.Get(resGroup, serverName, name)
+	if err != nil {
+		if resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Sql Elastic Pool %s: %s", name, err)
+	}
+
+	elasticPool := *resp.ElasticPoolProperties
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+	d.Set("server_name", serverName)
+	d.Set("edition", string(elasticPool.Edition))
+	d.Set("dtu", int(*elasticPool.Dtu))
+	d.Set("db_dtu_min", int(*elasticPool.DatabaseDtuMin))
```

Comment:
is it worth checking that `resp.ElasticPoolProperties` isn't `nil` before assigning these, to prevent a possible crash?